codalab/apps/teams/forms.py

```diff
@@ -89,3 +95,144 @@ def clean(self):
+                                              code='invalid')
+            else:
+                existing_team_with_name = Team.objects.filter(name=self.cleaned_data.get('name'))
+                if existing_team_with_name:
+                    raise ValidationError(('Invalid value for name. A team already exists with that name.'), code='invalid')
+        return self.cleaned_data.get('name')
+
+    def save(self, commit=True):
+        self.instance.creator = self.creator
+        self.instance.competition = self.competition
+        self.instance.status = TeamStatus.objects.get(codename=TeamStatus.APPROVED)
+
+        # We need to call save before adding members
+        self.instance.save()
+
+        if hasattr(self, 'user_list'):
+            # On save, if we have existing objects, clear it. We only add names in the form.
+            if len(self.instance.members.all()) > 0:
+                # Remove users no longer in the list
+                for user in self.instance.members.all():
+                    if user not in self.user_list:
+                        # Delete members not on the new list
+                        print("Deleting membership for user: {}".format(user))
+                        TeamMembership.objects.get(user=user, team=self.instance).delete()
+            # Then we loop through the names/emails they gave us.
+            for user in self.user_list:
+                if user not in self.instance.members.all():
+                    # There is not already an existing membership for this user.
+                    new_team_membership = TeamMembership.objects.create(
+                        user=user,
+                        team=self.instance,
+                        is_invitation=False,
+                        is_request=False,
+                        start_date=timezone.now(),
+                        end_date=timezone.now() + datetime.timedelta(days=9000),
+                        message="Organizer Created",
+                        status=TeamMembershipStatus.objects.get(codename=TeamMembershipStatus.APPROVED),
+                        reason="Organizer Created Team"
+                    )
+                    print("Created new membership: {0} for user: {1} on team: {2}".format(new_team_membership, user, self.instance))
+                else:
+                    print("Existing membership found for user on team.")
+        return super(OrganizerTeamForm, self).save(commit=commit)
+
+
+class OrganizerTeamsCSVForm(forms.Form):
+
+    csv_file = forms.FileField(required=False)
+
+    def __init__(self, *args, **kwargs):
+        self.competition = Competition.objects.get(pk=kwargs.pop('competition_pk'))
+        self.creator = ClUser.objects.get(pk=kwargs.pop('creator_pk'))
+        self.teams_dict = {}
+        super(OrganizerTeamsCSVForm, self).__init__(*args, **kwargs)
+
+    def clean_csv_file(self):
+        teams_dict = {}
```

Comment:
I don't think you need the `self.field` class attributes, just return the value a la https://docs.djangoproject.com/en/1.7/ref/forms/validation/#cleaning-a-specific-field-attribute